Service/RatingService.php

```diff
@@ -7,47 +7,139 @@
+use Symfony\Component\Console\Style\SymfonyStyle;
 
 class RatingService
 {
     private EntityManagerInterface $entityManager;
-    private GithubApiService $githubService;
+    private GithubApiService $githubApiService;
     private array $configuration;
     private array $data;
+    private ?Entity $componentEntity;
+    private ?Entity $ratingEntity;
+    private SymfonyStyle $io;
 
     public function __construct(
         EntityManagerInterface $entityManager,
-        GithubApiService $githubService
+        GithubApiService $githubApiService
     ) {
         $this->entityManager = $entityManager;
-        $this->githubService = $githubService;
+        $this->githubApiService = $githubApiService;
         $this->configuration = [];
         $this->data = [];
     }
 
     /**
-     * @param array $data          data set at the start of the handler
-     * @param array $configuration configuration of the action
+     * Set symfony style in order to output to the console.
      *
-     * @throws GuzzleException|Exception
+     * @param SymfonyStyle $io
      *
-     * @return array dataset at the end of the handler
+     * @return self
      */
-    public function enrichComponentWithRating(array $data, array $configuration): array
+    public function setStyle(SymfonyStyle $io): self
     {
-        $this->configuration = $configuration;
-        $this->data = $data;
+        $this->io = $io;
+        $this->githubApiService->setStyle($io);
 
-        $componentEntity = $this->entityManager->getRepository('App:Entity')->find($this->configuration['componentEntityId']);
-        $ratingEntity = $this->entityManager->getRepository('App:Entity')->find($this->configuration['ratingEntityId']);
+        return $this;
+    }
+
+    /**
+     * Get the component entity.
+     *
+     * @return ?Entity
+     */
+    public function getComponentEntity(): ?Entity
+    {
+        if (!$this->componentEntity = $this->entityManager->getRepository('App:Entity')->findOneBy(['reference'=>'https://opencatalogi.nl/oc.component.schema.json'])) {
+            isset($this->io) && $this->io->error('No entity found for https://opencatalogi.nl/oc.component.schema.json');
+        }
 
-        $this->rateComponent($data['request'], $ratingEntity);
+        return $this->componentEntity;
+    }
 
-//        foreach ($componentEntity->getObjectEntities() as $component) {
-//            $this->rateComponent($component, $ratingEntity);
-//        }
+    /**
+     * Get the rating entity.
+     *
+     * @return ?Entity
+     */
+    public function getRatingEntity(): ?Entity
+    {
+        if (!$this->ratingEntity = $this->entityManager->getRepository('App:Entity')->findOneBy(['reference'=>'https://opencatalogi.nl/oc.rating.schema.json'])) {
+            isset($this->io) && $this->io->error('No entity found for https://opencatalogi.nl/oc.rating.schema.json');
+        }
 
-        return $this->data;
+        return $this->ratingEntity;
```

Comment:
why return a $this variable?

Reply:
We currently do this in every service, it is something like a getter.
But i can agree that it should probably return null in case of an error